core/setup.py

```diff
@@ -14,17 +18,51 @@
 PACKAGE = "wandb_core"
 
 
-class WrapBdistWheel(bdist_wheel):
-    def get_tag(self):
-        # Use the default implementation to get python and abi tags
-        python, abi = bdist_wheel.get_tag(self)[:2]
-        # Use the wheel package function to determine platform tag
-        plat_name = get_platform(self.bdist_dir)
-        # todo: add MACOSX_DEPLOYMENT_TARGET to support older macs
-        return python, abi, plat_name
+class CustomBuildPy(build_py):
+    """Custom step to copy pre-built binary artifacts into the package."""
 
     def run(self):
-        bdist_wheel.run(self)
+        pkgdir = pathlib.Path(__file__).parent / PACKAGE
+
+        # Figure out the normalized architecture name for our current arch.
+        arch = platform.machine().lower()
+        if arch == "arm64":
+            arch = "aarch64"
+        elif arch == "amd64":
+            arch = "x86_64"
+
+        # We use cibuildwheel to create platform-specific wheels.
+        #
+        # On the ARM64 macOS-14 GitHub runner, platform.machine() sometimes
+        # returns x86_64 instead of arm64. This seems to be caused by
+        # cibuildwheel downloading an x86_64 Python on older machines, causing
+        # it to run via Rosetta, which (probably) causes `uname -m` to return
+        # x86_64.
+        #
+        # In these cases, `sysconfig.get_platform()` seems to still have the
+        # correct information.
+        sysplat = sysconfig.get_platform()
+        if sysplat.endswith("arm64"):
+            arch = "aarch64"
+            print(
+                f"setup.py: target architecture is '{arch}' "
+                f"(from sysconfig.get_platform() == '{sysplat}')"
+            )
+        else:
+            print(f"setup.py: target architecture is '{arch}'")
+
+        # Symlink the artifacts into bin/. The build system will copy the
+        # actual files into the wheel.
+        archdir = pkgdir.parent / "wandb_core_artifacts" / arch
+        for file in archdir.iterdir():
+            dest = pkgdir / file.name
+
+            if dest.exists():
+                dest.unlink()
+
+            os.symlink(file, dest)
```

Comment:
i think it should be fine for windows 10 and newer, but i think there used it a problem with symlinks for older windows systesm.

Reply:
do people develop on pre-Windows 10 machines? that's quite old, and I don't know if we have a way of testing on them

Reply:
yeah, i think it is fine, just wanted to let you know.. just in case..